### ISSUE TITLE
关于yolov8的实例分割自动标注问题

### ISSUE THREAD

**Issue by QzYER** [CVHub520/X-AnyLabeling#109]
官方onnx推理是正常的，用的你们提供的yaml文件，自动标注没有任何反应。

**Comment by CVHub520** [comment]
Hi, @QzYER:
thank you for your feedback, please package and send your ONNX file, YAML configuration file, and a sample image where you're certain the model should detect objects to the specified email address: [cv_hub@163.com](mailto:cv_hub@163.com).

I will do our best to assist you further.

Best regards,
CVHub

**Comment by CVHub520** [comment]
Hi, @QzYER, thank you for reaching out. I've tested the provided ONNX inference using the custom YAML file you provided, and everything seems to be working as expected. The automatic annotation process is responding as intended.

I would recommend trying the following:
1. Update your source code to the latest version.
2. Alternatively, you can directly use the most recent GUI version of the tool, which might include bug fixes and improvements.

Feel free to reach out if the issue persists or if you have any further questions.

**Comment by QzYER** [comment]
Last night, I retrained the YOLOV8X-SEG model until overfitting; Compared with the previous ONNX model, it was found that the accuracy of instance segmentation and converting pt files to ONNX files would decrease, resulting in unsatisfactory segmentation results during the automatic annotation process of custom models.
I hope to directly support the inference of pt files in later versions.Finally, thank you for your open source work.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "CVHub520/X-AnyLabeling"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年11月13日(星期一) 晚上8:26
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [CVHub520/X-AnyLabeling] 关于yolov8的实例分割自动标注问题 (Issue #109)





 
Hi, @QzYER, thank you for reaching out. I've tested the provided ONNX inference using the custom YAML file you provided, and everything seems to be working as expected. The automatic annotation process is responding as intended.
 
I would recommend trying the following:
  
Update your source code to the latest version.
 
Alternatively, you can directly use the most recent GUI version of the tool, which might include bug fixes and improvements.
  
Feel free to reach out if the issue persists or if you have any further questions.

**Comment by CVHub520** [comment]
Hi, @QzYER , thank you for your feedback, and I appreciate your recognition of this open-source efforts.

The observed differences between the YOLOv8X-SEG model and the ONNX conversion might arise from variations in some steps. For instance, in `X-AnyLabeling`, it utilize a straightforward resize method for preprocessing, while the original training framework might use letterboxing. It's worth noting that the tool's model inference backend is based on onnxruntime, chosen for its efficiency compared to the bulkier and redundant torch library.

If you have specific accuracy requirements and prefer a direct inference from `.pt` files, you can consider extending the [Model](https://github.com/CVHub520/X-AnyLabeling/blob/main/anylabeling/services/auto_labeling/model.py) class and implementing a custom PyTorch-based inference logic.

Feel free to reach out if you have further questions or need assistance. Your input is valuable to me!
